public/app/features/alerting/unified/components/receivers/TemplatesTable.tsx

```diff
@@ -27,7 +28,15 @@ export const TemplatesTable: FC<Props> = ({ config, alertManagerName }) => {
   const tableStyles = useStyles2(getAlertTableStyles);
   const permissions = getNotificationsPermissions(alertManagerName);
 
-  const templateRows = useMemo(() => Object.entries(config.template_files), [config]);
+  const templateRows = useMemo(() => {
+    const templates = Object.entries(config.template_files);
+
+    return templates.map(([name, template]) => ({
+      name,
+      template,
+      provenance: (config.template_file_provenances ?? {})[name],
```

Comment:
Same question as above. Wouldn't be better to join provenance with the rest of the template data on a higher level? I think we could avoid glue code spread in multiple places

Reply:
Another option is to create selectors that handle the glue code